types/src/front/lib/core_api.ts

```diff
@@ -250,7 +254,15 @@ export class CoreAPI {
     inputs,
     config,
     credentials,
+    secrets,
   }: CoreAPICreateRunParams): Promise<CoreAPIResponse<{ run: CoreAPIRun }>> {
+
+    // Convert the secrets array to an object for easy use by Core API with env.secrets.SECRET_NAME.
+    const secretsObject: SecretsObject = secrets.reduce((obj: SecretsObject, secret: DustAppSecretType) => {
```

Comment:
Why do we need that gymnastic. It looks out of place?

Reply:
Indeed. Moved it to Rust as this is where we need to move it to a HashMap to use {env.secrets.NAME} instead of an array.